wsdialer/wsdialer.go

```diff
@@ -192,9 +223,31 @@ func (dialer *WsDialer) Close() error {
 	if dialer == nil {
 		return nil
 	}
+	dialer.closeLock.Lock()
+	defer dialer.closeLock.Unlock()
+	if dialer.closed != nil {
+		defer close(dialer.closed)
+		dialer.closed = nil
```

Comment:
Doesn't closing nil channel panic?

Reply:
it would, however the value of closed is evaluated when the defer is called, so it's not nil. This would panic:
`defer func() {close(dialer.closed)}()`, since it's not evaluated until the defer.

Reply:
small example to demonstrate: https://play.golang.com/p/AqXnRN_k0Bu

Reply:
right, it is not nil when evaluated